pupepat/fitting.py

```diff
@@ -135,81 +133,97 @@ def fit_cutout(data, source, plot_filename, image_filename, header, id, fit_circ
     brightness_guess = np.median(cutout[np.logical_and(r < outer_radius_guess, r > inner_radius_guess)])
     inner_brightness_guess = np.median(cutout[r < inner_radius_guess])
 
-    initial_model = elliptical_annulus(x0_inner=x0, y0_inner=y0,
-                                       x0_outer=x0, y0_outer=y0,
-                                       amplitude_inner=inner_brightness_guess, amplitude_outer=brightness_guess,
-                                       a_inner=inner_radius_guess, b_inner=inner_radius_guess,
-                                       a_outer=outer_radius_guess,
-                                       b_outer=outer_radius_guess,
-                                       background=np.median(data))
-
-    if not fit_gradient:
-        initial_model.x_slope.fixed = True
-        initial_model.y_slope.fixed = True
-
+    x, y = np.meshgrid(np.arange(cutout.shape[1]), np.arange(cutout.shape[0]))
+    error = np.sqrt(np.abs(cutout))
     if fit_circle:
-        initial_model.theta_inner.fixed = True
-        initial_model.theta_outer.fixed = True
-        initial_model.b_inner.tied = lambda x: x.a_inner
-        initial_model.b_outer.tied = lambda x: x.a_outer
+        best_fit_parameter_names = ['x0_inner', 'y0_inner', 'r_inner', 'amplitude_inner',
+                                    'x0_outer', 'y0_outer', 'r_outer', 'amplitude_outer', 'background']
+        best_fit_solution = minimize(lambda *args: -ln_likelihood_circle(*args),
+                                     [x0, y0, inner_radius_guess, inner_brightness_guess,
+                                      x0, y0, outer_radius_guess, brightness_guess, np.median(data)],
```

Comment:
No need to recalculate background from `np.median(data)` as it is already done on line 126. But in general might it be better to estimate the background from something like `np.median(cutout[r>2*outer_radius_guess])` or does it not really matter?

Reply:
I think median of data is good enough.